Destination/Backup.php

```diff
@@ -0,0 +1,85 @@
+<?php
+
+namespace Zenstruck\BackupBundle\Destination;
+use DateTime;
+
+/**
+ * Class Backup
+ *
+ * Abstraction of one backup file.
+ *
+ * @package Zenstruck\BackupBundle\Destination
+ */
+final class Backup
+{
+    /**
+     * @var string
+     */
+    private $key;
+
+    /**
+     * @var string
+     */
+    private $filename;
+
+    /**
+     * @var integer
+     */
+    private $size;
+
+    /**
+     * @var \DateTimeInterface
+     */
+    private $createdAt;
+
+    public function __construct($key, $filename, $size, $createdAt)
+    {
+        $this->key = $key;
+        $this->filename = $filename;
+        $this->size = $size;
+        $this->createdAt = is_integer($createdAt) ? date_timestamp_set(new DateTime(), $createdAt) : $createdAt;
+    }
+
+    /**
+     * @return string
+     */
+    public function getKey()
+    {
+        return $this->key;
+    }
+
+    /**
+     * @return string
+     */
+    public function getFilename()
+    {
+        return $this->filename;
+    }
+
+    /**
+     * @return int
+     */
+    public function getSize()
```

Comment:
Me to, but people tends to avoid dependencies, so therefore - both are used - it will try library that you have provided, and if it can not be found, it will fall back to Util class.

In composer.json we will put library that you recommended under suggest section, and state that in documentation.

Util class will cover use cases like, 100MB, 100GB, 1TB when configuring rotators -> which is 99%. of use cases. 

You can not make someone to pull whole library for only one functionality which can fit into one method. :)


> On Sep 11, 2015, at 4:22 PM, Kevin Bond <notifications@github.com> wrote:
> 
> In Destination/Backup.php <https://github.com/kbond/ZenstruckBackupBundle/pull/3#discussion_r39275574>:
> 
> > +    {
> > +        return $this->key;
> > +    }
> > +
> > +    /**
> > +     * @return string
> > +     */
> > +    public function getFilename()
> > +    {
> > +        return $this->filename;
> > +    }
> > +
> > +    /**
> > +     * @return int
> > +     */
> > +    public function getSize()
> Ok, fair point.
> 
> I still think we should use the 3rd party library for parsing the $size from config rather than a Util class.
> 
> —
> Reply to this email directly or view it on GitHub <https://github.com/kbond/ZenstruckBackupBundle/pull/3/files#r39275574>.
>